tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -50,82 +50,93 @@ astDeclarationsForLanguage language filePath = do
 syntaxDatatype :: Ptr TS.Language -> Datatype -> Q [Dec]
 syntaxDatatype language datatype = case datatype of
   SumType (DatatypeName datatypeName) _ subtypes -> do
-    cons <- traverse (constructorForSumChoice datatypeName) subtypes
-    result <- symbolMatchingInstanceForSums language name subtypes
-    pure $ generatedDatatype name cons:result
+    typeParameterName <- newName "a"
```

Comment:
🗒 We’re constructing this in every branch, which suggests you could factor it out of this `case` expression, if you preferred.

Reply:
Good call, I did this. However, since these programs are built inside the quotation monad `Q` and `newName` was being used to generate an uncapturable name inside the monad, it is now using `mkName` giving us exactly the prescribed names instead of fresh identifiers.